Code/Tools/AssetProcessor/native/ui/BuilderData.cpp

```diff
@@ -0,0 +1,154 @@
+    {
+    }
+
+    //! This method runs when this model is initialized. It gets the list of builders, gets existing stats about analysis jobs and
+    //! processing jobs, and matches stats with builders and save them appropriately for future use.
+    void BuilderData::Reset()
+    {
+        BuilderInfoList builders;
+        AssetBuilderInfoBus::Broadcast(&AssetBuilderInfoBus::Events::GetAllBuildersInfo, builders);
+
+        m_root.reset(new BuilderDataItem(BuilderDataItem::ItemType::InvisibleRoot, "", 0, 0, nullptr));
+
+        m_allBuildersMetrics.reset(new BuilderDataItem(BuilderDataItem::ItemType::Builder, "All builders", 0, 0, m_root));
+        m_singleBuilderMetrics.clear();
+        m_builderGuidToIndex.clear();
+        m_builderNameToIndex.clear();
+        m_currentSelectedBuilderIndex = -1;
+
+        for (int i = 0; i < builders.size(); ++i)
+        {
+            m_singleBuilderMetrics.emplace_back(
+                new BuilderDataItem(BuilderDataItem::ItemType::Builder, builders[i].m_name, 0, 0, m_root));
+            m_builderGuidToIndex[builders[i].m_busId] = i;
+            m_builderNameToIndex[builders[i].m_name] = i;
+        }
+
+        // Analysis jobs stat
+        m_dbConnection->QueryStatLikeStatName(
+            "CreateJobs,%",
+            [this](AzToolsFramework::AssetDatabase::StatDatabaseEntry entry)
+            {
+                AZStd::vector<AZStd::string> tokens;
+                AZ::StringFunc::Tokenize(entry.m_statName, tokens, ',');
+                if (tokens.size() == 3) // CreateJobs,filePath,builderName
+                {
+                    const auto& sourceName = tokens[1];
+                    const auto& builderName = tokens[2];
+                    if (m_builderNameToIndex.contains(builderName))
+                    {
+                        m_singleBuilderMetrics[m_builderNameToIndex[builderName]]->UpdateOrInsertEntry(
+                            BuilderDataItem::JobType::CreateJob, sourceName, 1, entry.m_statValue);
+                        m_allBuildersMetrics->UpdateOrInsertEntry(
+                            BuilderDataItem::JobType::CreateJob, builderName + "," + sourceName, 1, entry.m_statValue);
+                    }
+                    else
+                    {
+                        AZ_Warning("AssetProcessor", false, "No builder found for an analysis job stat!!!\n");
+                    }
+                }
+                return true;
+            });
+
+        // Processing job stat
+        m_dbConnection->QueryStatLikeStatName(
+            "ProcessJob,%",
+            [this](AzToolsFramework::AssetDatabase::StatDatabaseEntry stat)
+            {
+                AZStd::vector<AZStd::string> tokens;
+                AZ::StringFunc::Tokenize(stat.m_statName, tokens, ',');
+                if (tokens.size() == 5) // ProcessJob,sourceName,jobKey,platform,builderGuid
+                {
+                    // const auto& sourceName = tokens[1];
```

Comment:
please remove commented out code

Reply:
Sorry, I forgot to check these thoroughly before making the PR. I will check them entirely and make an update.

Reply:
resolved